src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -95,7 +95,7 @@ public SafeDeleteSslContext(SslAuthenticationOptions sslAuthenticationOptions)
                 throw;
             }
 
-            if (!string.IsNullOrEmpty(sslAuthenticationOptions.TargetHost) && !sslAuthenticationOptions.IsServer)
+            if (!string.IsNullOrEmpty(sslAuthenticationOptions.TargetHost) && !sslAuthenticationOptions.IsServer && !!TargetHostNameHelper.IsValidAddress(sslAuthenticationOptions.TargetHost))
```

Comment:
The other changes only include one negation `!`, should this here also be just one?

```suggestion
            if (!string.IsNullOrEmpty(sslAuthenticationOptions.TargetHost) && !sslAuthenticationOptions.IsServer && !TargetHostNameHelper.IsValidAddress(sslAuthenticationOptions.TargetHost))
```